dotcom-rendering/index.d.ts

```diff
@@ -574,7 +574,7 @@ declare namespace JSX {
 	interface IntrinsicElements {
 		'gu-island': {
 			name: string;
-			deferUntil?: 'idle' | 'visible';
+			deferUntil?: 'idle' | 'visible' | 'interaction';
```

Comment:
`clicked`?

I think if you pressed me I would admit to preferring `interaction` but I'm putting this out there because maybe this works better 🤷 ?

Reply:
I'd second `clicked` because there are multiple kinds of interaction and click events are the kind that are actually being invoked here, e.g. tabbing to focus an `<input>` won't trigger a click event (afaik)

Reply:
I'd actually change the question to which forms of interaction _should_ we support, and do we want to support multiple, e.g some kind of interface

```tsx
<island
    deferUntil="interaction"
    events={["click", "keypress"]}
>
    ....
</island>
```

Reply:
this sounds good to me! I was thinking along similar lines to your `events` idea but I was wary of suggesting a premature abstraction; if you think we're likely to have use-cases for other event types any time soon then I agree that this is the way to go!

Reply:
To avoid premature abstraction, I'm going to keep it as on="interaction", as things like hitting enter on a button will cause it to hydrate (technically it's a click event, which is somewhat counter-intuitive). Definitely like the idea of the `events` api for the future though